### PR TITLE
feat: run specific migration file with --name option

### LINKED ISSUE #860: Run Specific migration file
I can undo a specific migration file using `sequelize db:migrate:undo --name migration_file_name`.
I wonder why there is no option to migration just specific migration file?
Why below doesn't work?
`sequelize db:migrate --name migration_file_name`
It's just common sense everyone will go to this If they found out the method to undo a specific migration.

### LINKED ISSUE #1028: Sequelize migrations not working
## What you are doing?
Running cli with `--name` to undo specific migrations does not work anymore.

```js
npx sequelize-cli db:migrate:undo —-name 20220215204453-create-table

Sequelize CLI [Node: 14.18.1, CLI: 6.4.1, ORM: 6.14.1]

Options:
  --version          Show version number                                                                  [boolean]
  --help             Show help                                                                            [boolean]
  --env              The environment to run the command in                        [string] [default: "development"]
  --config           The path to the config file                                                           [string]
  --options-path     The path to a JSON file with additional options                                       [string]
  --migrations-path  The path to the migrations folder                             [string] [default: "migrations"]
  --seeders-path     The path to the seeders folder                                   [string] [default: "seeders"]
  --models-path      The path to the models folder                                     [string] [default: "models"]
  --url              The database connection string to use. Alternative to using --config files            [string]
  --debug            When available show various debug information                       [boolean] [default: false]
  --name             Name of the migration to undo                                                         [string]

Unknown arguments: —-name, 20220215204453-create-table
```

## What do you expect to happen?
The migration undo runs smoothly

## What is actually happening?
I get an error that says `unknown arguments`

### PR DESCRIPTION
**Pull request by remarkablemark**
Closes #860, closes #1028

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

### Description of change

feat: run specific migration file with `--name` option

Example:

```sh
npx sequelize db:migrate --name 20111117063700-createPerson
```

### CODE REVIEW

File: src/commands/migrate.js [hunk 1]
```diff
@@ -17,6 +17,10 @@ exports.builder = (yargs) =>
     .option('from', {
       describe: 'Migration name to start migrations from (excluding)',
       type: 'string',
+    })
+    .option('name', {
+      describe: 'Migration name',
```
Comment (ephys) [inline src/commands/migrate.js:22]:
```suggestion
      describe: 'Migration name. When specified, only this migration will be run',
```

We should document that it's mutually exclusive with `to` & `from`.
We should also throw an error if these mutually exclusive options are provided together.

umzug can accept an array of migration name, we should make sure `name` can be specified more than once:

`sequelize-cli db:migrate --name migration1.js --name migration2.js`

**Review by ephys — CHANGES_REQUESTED** [review]
Looking good and thank you for adding tests :)

File: src/commands/migrate.js (hunk 1 above)
Comment (remarkablemark) [reply src/commands/migrate.js]: c22a291

**Comment by WikiRik** [comment]
Hi! Thanks for your work on this, could you also add a little bit about this to the [main sequelize documentation](https://github.com/sequelize/sequelize/blob/main/docs/manual/other-topics/migrations.md)? And maybe add a little bit to the description of the command here so it's clear to people that they can specify multiple migrations by repeating single migrations with the `--name` flag?

And as a bonus feature, can we make it so that `--name` accepts an array as well?

**Review by ephys — APPROVED** [review]
Looks good to me!

@WikiRik by array do you mean [this](https://github.com/sequelize/cli/pull/1034/files#diff-18525de583bcba92a422af366d5c414b0882d1d14e9092d07a398f89f6037498R610) or an alternative syntax?

**Comment by WikiRik** [comment]
On mobile your link does not go anywhere unfortunately, but I was thinking about `--name createPersonMigration, createPostMigration` or something like that. It's a bit shorter than `--name createPersonMigration --name createPostMigration`.
But as I said, that's just a bonus. I think we can also merge without that and leave this for a future PR

**Comment by ephys** [comment]
@WikiRik just checking: is this good for you? Can I merge?

**Comment by WikiRik** [comment]
I don't see the comment anymore but we can merge this and add additional syntax options in a future PR

**Comment by ephys** [comment]
I deleted my comment because I saw you already responded in the previous one 😅

**Comment by adarsh-kumar1** [comment]
Not running as of now with single command using --
name

**Comment by WikiRik** [comment]
True, we have merged this but not released a new version yet. We will do that in the coming days

**Comment by SupunSPWithana** [comment]
> True, we have merged this but not released a new version yet. We will do that in the coming days

@WikiRik is this released now?

**Comment by WikiRik** [comment]
> is this released now?

Yes, the latest v6 release includes this

**Pull request merged**